Applications/Spire/Source/Ui/DropDownWindow.cpp

```diff
@@ -0,0 +1,131 @@
+#include "Spire/Ui/DropDownWindow.hpp"
+#include <QHBoxLayout>
+#include <QKeyEvent>
+#include <QPainter>
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace Spire;
+
+DropDownWindow::DropDownWindow(bool is_click_activated, QWidget* parent)
+    : QWidget(parent, Qt::Tool | Qt::FramelessWindowHint),
+      m_widget(nullptr),
+      m_is_click_activated(is_click_activated) {
+  setAttribute(Qt::WA_ShowWithoutActivating);
+  setAttribute(Qt::WA_TranslucentBackground);
+  m_shadow = new DropShadow(true, false, this);
+  auto layout = new QHBoxLayout(this);
+  layout->setContentsMargins(scale_width(1), scale_height(1),
+    scale_width(1), scale_height(1));
+  parent->installEventFilter(this);
+  parent->window()->installEventFilter(this);
+  hide();
+}
+
+bool DropDownWindow::event(QEvent* event) {
+  if(event->type() == QEvent::WindowDeactivate &&
+      focusWidget() != nullptr &&
+      !isAncestorOf(focusWidget())) {
+    hide();
+  }
+  return QWidget::event(event);
+}
+
+bool DropDownWindow::eventFilter(QObject* watched, QEvent* event) {
+  if(watched == m_widget) {
+    if(event->type() == QEvent::Resize) {
+      resize(m_widget->width() + 2, m_widget->height() + 2);
```

Comment:
Briefly explain this, why 2 pixels? Why to both the width and height?

Reply:
The DropDownWindow has a border on all sides, so it's an allowance to have the border visible.

Reply:
Makes sense.